### ISSUE TITLE
The "custom region demo" doesn't do what it says it does

### ISSUE THREAD

**Issue by mrcslws** [numenta/nupic#3031]
There are a couple of issues in:

/examples/network/custom_region/identity_region.py
/examples/network/custom_region_demo.py

## Issue 1

It says it passes "the encoded representation to an Identity Region". But it actually passes the unencoded input. 

This line uses the wrong source on the RecordSensor:

~~~python
  network.link("sensor", "identityRegion", "UniformLink", "",
               srcOutput="sourceOut", destInput="in")
~~~

It should use srcOutput "dataOut", or just use the default.

~~~python
  network.link("sensor", "identityRegion", "UniformLink", "")
~~~

## Issue 2

The IdentityRegion is not actually writing to the output. It's reassigning the output variable to reference a totally different array.

The fix: Change

~~~python
outputs["out"] = inputs["in"]
~~~

to

~~~python
outputs["out"][:] = inputs["in"]
~~~

## Issue 3

Now that we've solved those, we need to put the correct `dataWidth` in the IdentityRegion.

Basically, change the line:

~~~python
network.addRegion("identityRegion", "py.IdentityRegion", json.dumps(I_PARAMS))
~~~

to:

~~~python
network.addRegion("identityRegion", "py.IdentityRegion", json.dumps({
        "dataWidth": sensor.encoder.getWidth(),
    }))
~~~
